config/src/genesis.rs

```diff
@@ -73,16 +73,19 @@ impl Configuration {
                     raw_block,
                 })
             }
-            _ => Err(ParseError::InvalidParametersCombination),
+            (_, _, true) => Err(ParseError::SubmitIsSetButRestAreNot),
+            (_, _, false) => Err(ParseError::SubmitIsNotSetButRestAre),
         }
     }
 }
 
 /// Error which might occur during [`Configuration::parse()`]
 #[derive(Debug, displaydoc::Display, thiserror::Error)]
 pub enum ParseError {
-    /// `genesis.private_key` and `genesis.file` should be set for the peer that submits the genesis
-    InvalidParametersCombination,
+    /// `--submit-genesis` was provided, but `genesis.private_key` and/or `genesis.file` are missing
+    SubmitIsSetButRestAreNot,
+    /// `--submit-genesis` was not provided, but `genesis.private_key` and/or `genesis.file` are set
+    SubmitIsNotSetButRestAre,
```

Comment:
Can we refactor it in that style?

```rust
pub enum SubmitSetError {
    PrivateKeyMissing,
    FileMissing,
}
```

I just don't like to provide uncertainty when we can capture the exact error

Reply:
To be exact, there are 6 possible error combinations:

| `--submit-genesis` | `genesis.private_key` | `genesis.file` |
| :-: | :-: | :-: |
| `true` | None | None |
| `true` | Some | None |
| `true` | None | Some |
| `false` | Some | None |
| `false` | None | Some |
| `false` | Some | Some |

It would be great to tell the user about the exact error, I agree. However, I feel reluctant to be very detailed on this stage of config overhaul, since genesis design & config might be changed soon, and I am anyway going to completely re-write `iroha_config` as a next stage, which will also address errors detalisation, not only in genesis.

Reply:
Fair enough